common/log/zap_logger.go

```diff
@@ -174,6 +186,24 @@ func (l *zapLogger) Skip(extraSkip int) Logger {
 	}
 }
 
+func mergeTags(oldTags, newTags []tag.Tag) (tags []tag.Tag) {
+	if oldTags == nil {
+		return newTags
+	}
+	tagMap := make(map[string]tag.Tag, len(oldTags)+len(newTags))
```

Comment:
there's usually only like 1-10 tags and With calls add a small number.. maybe we should just do a linear search instead? using a map like this will shuffle the order, which isn't great since related tags are often passed together.

or else use the map differently: clone the old slice, make a map from key to index, and then lookup and replace or else append?

Reply:
I went with plan A